thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/auto/onboard/ConfigGenerator.java

```diff
@@ -92,6 +95,16 @@ private static TimeGranularity getExpectedDelayFromTimeunit(TimeUnit timeUnit) {
     return expectedDelay;
   }
 
+  /**
+   * Check if the dataset is non-additive. If it is, set the additive flag to false and set the pre-aggregated keyword.
+   * @param dataset the dataset DTO to check
+   */
+  static void checkNonAdditive(DatasetConfigDTO dataset) {
+    if (dataset.isAdditive() && dataset.getDataset().endsWith(NON_ADDITIVE)) {
```

Comment:
I don't understand the logic here.
So by default the dataset is additive()?
If so then why we need this dataset.isAdditive() check?
If it is one-time fix suggest not to put it into the onboard logic.

Reply:
Actually, it's not a one-time fix. The logic a bit tricky here.

When creating a `new DatasetConfigDTO()`, the additive is `true`. Then we check if it's a non_additive dataset, we set the flag to be `false` and set the keyword to be `*`.

When checking an existing  `DatasetConfigDTO`, if additive = `true` for a non_additive dataset, we set the flag to be `false` and set the keyword to be `*`.  But if additive = `false` for a non_additive dataset, we leave the flag and the keyword unchanged. If remove the `dataset.isAdditive()` check, we might override the `PreAggregatedKeyword`. For some old dataset, the keyword is set to `ALL` instead of `*`.